Examples/DicomConvert/DicomConvert.cxx

```diff
@@ -0,0 +1,181 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+/**
+ * A SimpleITK example demonstrating how to convert and resize DICOM files
+ * to common image types.
+ */
+
+#include <SimpleITK.h>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <filesystem>
+
+namespace sitk = itk::simple;
+namespace fs = std::filesystem;
+
+bool
+ConvertImage(const std::string & inputFile, const std::string & outputFile, unsigned int newWidth = 0)
+{
+  try
+  {
+    sitk::ImageFileReader reader;
+    // Only read DICOM images
+    reader.SetImageIO("GDCMImageIO");
+    reader.SetFileName(inputFile);
+    reader.ReadImageInformation();
+    auto                      iSize = reader.GetSize();
+    std::vector<unsigned int> imageSize;
+    imageSize.push_back(static_cast<unsigned int>(iSize[0]));
+    imageSize.push_back(static_cast<unsigned int>(iSize[1]));
+    // If 3D image with single slice, treat as 2D
+    if (iSize.size() == 3)
+    {
+      if (iSize[2] == 1)
+      {
+        imageSize.push_back(0U);
+      }
+      else
+      {
+        imageSize.push_back(static_cast<unsigned int>(iSize[2]));
+      }
+    }
+    reader.SetExtractSize(imageSize);
```

Comment:
You are always reading the whole image. All these conditional are just to convert a single size image to 3D->2D.  I would just read the image. Check is dimension, then if dim==3 then use the extract image filter to reduce the dimension. This should be more compact and skip this messiness with types.

Reply:
Sounds good, will replace the usage of the reader's setExtractSize with the extract image filter in the strongly typed languages.